### ISSUE TITLE
Account sessionInfo is null

### ISSUE THREAD

**Issue by gray419** [SAP/gigya-flutter-plugin#5]
Hey all, when calling:
```dart
final response = await GigyaSdk.instance.login();
final account = Account.fromJson(response);
```
I'm seeing the [sessionInfo](https://github.com/SAP/gigya-flutter-plugin/blob/main/lib/models/gigya_models.dart#L107-L109) property as null.  When I inspect the network response - I see this sessionInfo in the response, but it seems to not de-serialize succesfully.

<img width="1072" alt="Screen Shot 2021-04-14 at 2 29 29 PM" src="https://user-images.githubusercontent.com/3689237/114760937-e4a46400-9d2d-11eb-9014-d71b333b12c3.png">

**Comment by tal-mi** [comment]
Hi,

Please follow this flow as described in example project:
the login call is async.

```
GigyaSdk.instance.login(loginId, password).then((result) {
      final response = Account.fromJson(result);
      setState(() {
        _inProgress = false;
        _requestResult = 'Login success:\n\n ${response.uid}';
      });
    }).catchError((error) {
      if (error.getInterruption() == Interruption.conflictingAccounts) {
        LinkAccountResolver resolver = GigyaSdk.instance.resolverFactory.getResolver(error);
        _resolveLinkAccount(resolver);
      } else {
        setState(() {
          _inProgress = false;
          _requestResult = 'Register error\n\n${error.errorDetails}';
        });
      }
    });
```
 
Regards,
Tal

**Comment by gray419** [comment]
Hey @tal-mi thanks for the response. I've updated my code snippet to show that I'm awaiting the async call and still not seeing the sessionInfo property. This happens in the example app as well.

**Comment by tal-mi** [comment]
Hi,
That's odd. I've just verified it (before I answered) on the example application with the latest version.
Can you send the exact code snippet you are using and which version/platform?

Thanks,
Tal

**Comment by gray419** [comment]
Here are the reproduction steps:

- Fetch latest from main branch, commit `0c126bfe448526b5ebb813e7c60eb8365270a1cb`
- Updated GigyaApiKey in the Info.plist to point to our Gigya site
- updated `_sendLoginRequest` method print the sessionInfo.sessionToken here:

```dart
  void _sendLoginRequest(loginId, password) async {
    setState(() {
      _inProgress = true;
    });
    GigyaSdk.instance.login(loginId, password).then((result) {
      debugPrint(json.encode(result));
      final response = Account.fromJson(result);
      print('sessionToken: ${response.sessionInfo?.sessionToken ?? 'null'}');
      setState(() {
        _inProgress = false;
        _requestResult = 'Login success:\n\n ${response.uid}';
      });
    }).catchError((error) {
      if (error.getInterruption() == Interruption.conflictingAccounts) {
        LinkAccountResolver resolver =
            GigyaSdk.instance.resolverFactory.getResolver(error);
        _resolveLinkAccount(resolver);
      } else {
        setState(() {
          _inProgress = false;
          _requestResult = 'Register error\n\n${error.errorDetails}';
        });
      }
    });
  }
```

The log output 
```
flutter: {"isActive":true,"isVerified":false,"UID":"d533b1019d2d4a6fa1fb7077c8554846","isRegistered":true,"registeredTimestamp":1610991489,"registered":"2021-01-18T17:38:09.138Z","createdTimestamp":1610991488,"signatureTimestamp":"1618756173","oldestDataUpdatedTimestamp":1610991488961,"socialProviders":"site","lastUpdated":"2021-04-07T23:05:21.659Z","oldestDataUpdated":"2021-01-18T17:38:08.961Z","profile":{"firstName":"RobDog","email":"rob.g+30@test.com","lastName":"","zip":"11222"},"lastLoginTimestamp":1618756173,"created":"2021-01-18T17:38:08.961Z","loginProvider":"site","lastLogin":"2021-04-18T14:29:33.333Z","UIDSignature":"0h9WgB7ue2u1GFW7Foe+sdhwphA=","lastUpdatedTimestamp":1617836721659,"apiVersion":2}
flutter: sessionToken: null
```

<img width="1920" alt="Screen Shot 2021-04-18 at 10 37 39 AM" src="https://user-images.githubusercontent.com/3689237/115149440-2422c700-a032-11eb-8f47-877e340db4c8.png">

**Comment by tal-mi** [comment]
Thank you.
We will investigate and update.

**Comment by tal-mi** [comment]
The issue will be resolved in 0.0.6 release.

In order to get the "sessionInfo" object, you will need to add the following to your custom account class in your host app:
var sessionInfo: [String: AnyCodable]?

See UserHost.swift class in the example project.

In addition, we have fixed our account model to adjust the "sessionExpiration" getter to match both core SDK serialization.

Regards,
Tal
